crates/api/src/wasm.rs

```diff
@@ -122,8 +122,8 @@ macro_rules! declare_vec {
             }
 
             #[allow(dead_code)]
-            fn as_slice(&self) -> &[$elem_ty] {
-                unsafe { slice::from_raw_parts(self.data as *const $elem_ty, self.size) }
+            fn as_slice(&self) -> &[*mut $elem_ty] {
```

Comment:
Here the cast was copied from above, removing a level of pointer-indirection by mistake, resulting in a corrupted slice.

Likewise, the cast is unnecessary and the element type should be `*mut T`.